lib/engine/game/g_18_india/step/certificate_selection.rb

```diff
@@ -55,57 +58,75 @@ def players_visible?
             false
           end
 
+          def show_companies
+            true
+          end
+
+          def show_map
+            true
+          end
+
           def name
-            'Hand Selection'
+            'Initial Hand Selection'
           end
 
           def description
             "Select #{@cards_to_keep} Certificates for your starting hand"
           end
 
-          def finished?
-            @confirmed_selections == @game.players.size
+          def selection_note
+            [
+              "Made #{number_of_selections} of #{@cards_to_keep} selections.",
+              'Click on card to select or unselect it.',
+            ]
           end
 
-          def selections_completed?
-            number_of_selections == @cards_to_keep
+          def select_company(player, company)
+            return if selections_completed? && company.owner.nil? # prevent selecting more than allowed
```

Comment:
It might be a better user interface to allow a player to select more items than they are allowed to keep, but not allow them to finish their selection until they have unselected other items. Otherwise it might not be obvious why nothing is happening when they are clicking on a share.

Reply:
Thanks for the suggestion, I will re-work it to allow selecting too many, but also change the NOTE to warn that too many are selected.

Reply:
removed check to allow selecting past limit.